# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.
diff --git a/output/csv/output.go b/output/csv/output.go
index 582ad7e5..c4e94ba2 100644
--- a/output/csv/output.go
+++ b/output/csv/output.go
@@ -198,7 +198,7 @@ func SampleToRow(sample *metrics.Sample, resTags []string, ignoredTags []string,
 	case TimeFormatRFC3339:
 		row[1] = sample.Time.Format(time.RFC3339)
 	case TimeFormatUnix:
-		row[1] = strconv.FormatInt(sample.Time.Unix(), 10)
+		row[1] = strconv.FormatInt(sample.Time.UnixMicro(), 10)
 	}

 	row[2] = fmt.Sprintf("%f", sample.Value)
